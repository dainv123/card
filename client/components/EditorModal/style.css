#editor-modal .card [data-schemapath] {
    display: flex;
}

#editor-modal .card .col-md-12[data-schemapath]>div {
    flex: 1;
}

#editor-modal .card .col-md-12[data-schemapath]>div [data-schematype="string"] .form-group,
#editor-modal .card .col-md-12[data-schemapath]>div [data-schematype="string"] .form-control {
    width: 100%;
}

#editor-modal .card [data-schemapath] .btn-secondary {
    font-size: 10px;
    padding: 4px 10px;
}

#editor-modal .card .form-control[aria-label] {
    width: 700px;
}

#editor-modal .card [data-schematype="array"] .form-control[aria-label] {
    width: 400px;
}

#editor-modal .card-title label,
#editor-modal .card [data-schemapath] label {
    min-width: 35px;
    text-transform: capitalize;
    font-weight: 600;
}

.viewport {
    overflow-y: auto;
    padding-right: 20px;
    scrollbar-color: rgba(27, 127, 204, .8) #e8e8e8;
    scrollbar-width: thin;
    max-height: calc(-347px + 100vh);
}

.viewport .item {
    display: flex;
    align-items: center;
}

.viewport .form-control.input-label {
    max-width: 120px;
    margin-right: 5px;
    background: #d9d9d9;
}

.viewport .form-control {
    display: block;
    width: 100%;
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    margin: 5px 0;
}

.viewport .scroller {
    max-height: calc(-349px + 100vh);
    overflow-y: auto;
}

.viewport .item-close {
    color: red;
    font-size: 30px;
    position: absolute;
    top: -10px;
    right: 3px;
    cursor: pointer;
}

/* .ant-modal.display-below .ant-modal-content {
    background-color: #ffffff94
} */

.margin-bottom-5 {
    margin-bottom: 5px;
}

.margin-right-5 {
    margin-right: 5px;
}
