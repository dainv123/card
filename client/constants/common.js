export const MY_NAME = 'Dai Nguyen';
export const COLOR_BLACK_1 = 'rgba(0,0,0,.25)';
export const MY_GITHUB = "https://github.com/dainv123"
export const LOGO_URI = '/public/images/logo2.png';
export const AVATAR_URI = '/public/assets/images/avatar.jpg';
export const ABOUT_ME_IMAGE = '/public/assets/images/about.png';
export const CONTACT_MAILBOX = '/public/assets/images/mailbox.png';
export const UPLOADS_FOLDER = 'uploads';