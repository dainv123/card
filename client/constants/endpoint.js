// export const CLIENT_PORT = "3000";
// export const SERVER_PORT = "8080";
// export const CLIENT_URI = "http://localhost:3000";
// export const SERVER_URI = "http://localhost:8080";
export const CLIENT_URI = "https://daidev-bc51fcbb2e2d.herokuapp.com";
export const SERVER_URI = "https://daidev-bc51fcbb2e2d.herokuapp.com";
export const GRAPHQL_URI = SERVER_URI + "/graphql";
export const UPLOAD_URI = SERVER_URI + "/api/upload";
export const UPLOAD_DELETE_URI = SERVER_URI + "/api/upload-delete";
export const BLOG_URI = CLIENT_URI + "/blog/";
export const THEME_URI = SERVER_URI + "/theme/";
