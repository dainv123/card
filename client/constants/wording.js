/*************************
 WORDINGS
*************************/
export const COPY_RIGHT = 'DAIDEV - developed by testofdai@gmail.com';
export const UPDATE_USER_SUCCESSFULLy = 'User updated successfully';
export const MAKE_YOUR_BLOG = 'Make Your Blog';
export const ADD_YOUR_BLOG = "Add Your Blog";
export const UPDATE_YOUR_BLOG = "Update Your Blog";
export const CANCEL = 'Cancel';
export const SUBMIT = 'Submit';
export const UNDO = 'Undo';
export const EDIT = 'Edit';
export const NAME = 'Name';
export const TREND = 'Trend';
export const INTRODUCTION = 'Introduction';
export const CONTENT = 'Content';
export const IMAGE = 'Image';
export const MAKE_YOUR_CARD = 'Make Your Card';
export const ADD_YOUR_CARD = "Add Your Card";
export const UPDATE_YOUR_CARD = "Update Your Card";
export const ADD_YOUR_IMAGE = "Add Your Image";
export const UPDATE_YOUR_IMAGE = "Update Your Image";
export const SELECT_THEME = 'Select Theme';
export const CONFIG = 'Config';
export const SETTING = 'Setting';
export const RESET_TO_DEFAULT = 'Reset to default';
export const AI_FORM = 'AI Form';
export const ENTER_REQUIREMENT = 'Enter your requirement';
export const EMAIL_SENT = 'Email Sent!';
export const UPLOAD = 'Upload';
export const LOG_IN = 'Log In';
export const DASHBOARD = 'Dashboard';
export const SIGN_OUT = 'Sign Out';
export const HOME = 'Home';
export const ABOUT_ME = 'About Me';
export const THEME = 'Theme';
export const BLOG = 'Blog';
export const CONTACT = 'Contact';
export const LOGGED_IN_SUCCESSFULLY = 'Logged in successfully';
export const INCORRECT_EMAIL_PASSWORD = 'Incorrect email or password.';
export const EMAIL = 'Email';
export const PASSWORD = 'Password';
export const FORGOT_PASSWORD = 'Forgot password';
export const REMEMBER_ME = 'Remember me';
export const OR = 'Or';
export const REGISTER_NOW = 'register now!';
export const EMAIL_HAS_ALREADY = 'Email has already been taken.';
export const USERNAME_HAS_ALREADY = 'Username has already been taken.';
export const REGISTER = 'Register';
export const USERNAME = 'Username';
export const CONFIRM_PASSWORD = "Confirm Password";
export const AGREE_LOGIN = 'I agree to FAKE Terms of Service';
export const ALREADY_HAVE_ACCOUNT = 'Already have an account?';
export const MAKE_YOUR_TAG = "Make Your Tag";
export const ADD_YOUR_TAG = "Add Your Tag";
export const UPDATE_YOUR_TAG = "Update Your Tag";
export const MAKE_YOUR_THEME = "Make Your Theme";
export const ADD_YOUR_THEME = "Add Your Theme";
export const UPDATE_YOUR_THEME = "Update Your Theme";
export const PATH = "Path";
export const SELECT_TAGS = "Select Tags";
export const GITHUB = "Github";
export const COPY_CLIPBOARD_OK = 'Text copied to clipboard';
export const COPY_CLIPBOARD_ERROR = 'Failed to copy text to clipboard';
export const ABOUT_ME_INTRO = "WHAT DO YOU WANT TO KNOW ABOUT ME?";
export const ABOUT_ME_ADDRESS = 'Ho Chi Minh City, Viet Nam';
export const ABOUT_ME_CONTENT_1 = "Hey there! I'm Nguyen Van Dai, a Developer calling Ho Chi Minh city home ☀️. I identify as a passionate developer, embracing the world of coding, open source, and the web platform ❤️.";
export const ABOUT_ME_CONTENT_2 = "Beyond my professional responsibilities, I actively engage in crafting and contributing to open source projects, a practice that not only broadens my knowledge but also aids in the advancement of fellow open source endeavors. Additionally, I take pleasure in sharing my technical insights ✍️ through my blog or";
export const ABOUT_ME_CONTENT_3 = "During my leisure hours, you'll likely find me backpacking, exploring mountains or beaches 🏖, or attending tech fairs in and around Ho Chi Minh city.";
export const BLOG_INTRO = "I share my news and blog";
export const CONTACT_INTRO = "NEED SOME HELP?";
export const MESSAGE = 'Message';
export const SEND_MESSAGE = 'Send Message';
export const WEB_DEVELOPER = 'Senior Web Developer';
export const MOBILE_DEVELOPER = 'Also, Mobile Developer';
export const BACKEND_DEVELOPER = 'Culi in Some Backend Languages';
export const AND = 'and';
export const CONTACT_ME = 'Contact Me';
export const GO_TO_ADMIN = 'Go to Admin';
export const PAGE_NOT_FOUND = '404 - Page Not Found';
export const USER_NAME = "User Name";
export const SAVE = 'Save';
export const PROFILE = 'Profile';
export const SOMETHING_WENT_WRONG = 'Something went wrong!';
export const SEND_CONTACT_SUCCESSFULLY = 'Your message has been successfully sent!';
export const THEME_INTRO = 'Feel free to take a brief look and utilize them as needed.';
export const ALL = 'All';
export const E_MAIL = 'E-mail';
export const PROJECT = 'Project';
export const VERIFY_USER_SUCCESSFULLY = 'Verify user account successfully';
export const IMAGES = 'IMAGE(S)';
export const ADD_IMAGE = 'ADD IMAGE';
export const CARDS = 'CARD(S)';
export const ADD_CARD = 'ADD CARD';
export const THEMES = 'THEME(S)';
export const ADD_THEME = 'ADD THEME';
export const TAGS = 'TAG(S)';
export const ADD_TAG = 'ADD TAG';
export const BLOGS = 'BLOG(S)';
export const ADD_BLOG = 'ADD BLOG';
/*************************
 MESSAGES
*************************/
export const ENTER_VALID_INFO_MESSAGE = `Please provide a valid email and password`;
export const ENTER_EMAIL_PASSWORD_MESSAGE = `Please enter your email and password.`;
export const MUST_BE_VALID_EMAIL_MESSAGE = 'Must be a valid email';
export const MUST_ACCEPT_TERM_MESSAGE = 'Must Accept Terms and Conditions';
export const MUST_AT_LEAST_ONE_MESSAGE = 'Must have at least one letter and one digit.';
export const ONLY_NUMBER_LETTER_MESSAGE = 'Must only contain letters and numbers';
export const PASSWORD_MUST_MATCH_MESSAGE = 'Passwords must match';
export const PASSWORD_CONFIRMATION_MESSAGE = 'Password confirmation';
export const REQUIRE_PROCESSING_MESSAGE = "This field is required for AI processing";
export const REQUIRE_REQUIREMENT_MESSAGE = 'Please enter your requirement!';
/*************************
 CALLER
*************************/
export const HELLO_NAME = name => `Hello, I'm ${name}`;
export const MIN_MESSAGE = min => `Must have at least ${min} characters`;
export const MAX_MESSAGE = max => `Cannot have more than ${max} characters`;
export const REQUIRED_MESSAGE = field => `${field} is required`;
export const CHECK_EMAIL_PLEASE = email => `Email sent to ${email}! Check your email and verify your account.`;
export const FILE_UPLOAD_FAILED = filename => `${filename} file upload failed.`;
export const MAXIMUM_UPLOAD_MESSAGE = maxCount => `Maximum ${maxCount} images allowed.`;