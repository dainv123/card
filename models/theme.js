import mongoose from 'mongoose';

const { Schema, model } = mongoose;

const themeSchema = new Schema(
  {
    name: String,
    path: String,
    tags: [
      {
        type: Schema.Types.ObjectId,
        ref: 'Tag'
      }
    ],
    image: String,
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
  },
  {
    timestamps: true
  }
);

const Theme = model('Theme', themeSchema);

export default Theme;
