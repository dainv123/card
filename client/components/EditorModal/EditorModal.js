import axios from 'axios';
import React, { useState, useEffect, useRef } from 'react';
import Scroller from '../../pages/Reader/Scroller';
import { Modal, Card, Form, Input, Button } from 'antd';
import { JSONEditor } from '@json-editor/json-editor';
import { 
  AI_FORM, 
  CANCEL, 
  ENTER_REQUIREMENT, 
  REQUIRE_PROCESSING_MESSAGE, 
  REQUIRE_REQUIREMENT_MESSAGE, 
  RESET_TO_DEFAULT, 
  SETTING, 
  SUBMIT,
} from '../../constants/wording';
import './style.css';

const EditorModal = ({ data = {}, isModalOpen, handleOk, handleCancel }) => {
  const aiRef = useRef(null);
  const editorRef = useRef(null);
  const [top, setTop] = useState('0px');
  const [left, setLeft] = useState('0px');
  const [node, setNode] = useState({ id: null, message: '', require: '' });
  const [isAI, setIsAI] = useState(false);
  const [editor, setEditor] = useState(null);
  const [isOpen, setIsOpen] = useState(false);

  const onReset = () => {
    if (editor) {
      handleOk({});
      editor.destroy();
    }
  };

  const onCancel = () => {
    if (editor) {
      editor.destroy();
    }

    handleCancel();
  };

  const onSubmit = () => {
    if (editor) {
      const editorValue = editor.getValue();

      for (const [key, value] of Object.entries(editorValue)) {
        editorValue[key] = { default: value };
      }

      handleOk(editorValue);

      editor.destroy();
    }
  };

  const onChangeAI = (e) => { 
    setNode({ id: node.id, message: node.message, require: e.target.value });
  };

  const onSubmitAI = (event) => {
    event.preventDefault();
    // const key = process.env.EMAIL_PASSWORD;
    // const apiKey = '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';
    // const apiUrl = 'https://api.openai.com/v1/chat/completions';
    // axios.post(
    //   apiUrl,
    //   {
    //     "model": "gpt-3.5-turbo",
    //     "messages": [{"role": "user", "content": `"${node.message}" ${node.require}`}],
    //     "temperature": 0.7
    //   },
    //   {
    //     headers: {
    //       Authorization: `Bearer ${apiKey}`,
    //       'Content-Type': 'application/json',
    //     },
    //   }
    // ).then(res => {
    //   const inputString = 'root[node_11]';
    //   const match = node.id.match(/\[([^[\]]+)\]/);

    //   if (match && match[1]) {
    //     const extractedContent = match[1];
    //     editor.setValue({ ...editor.getValue(), [extractedContent]: res.data.choices[0].message.content });
    //   }
    // })
  };

  useEffect(() => {
    setIsOpen(isModalOpen);
    setTimeout(() => {
      if (editor) {
        editor.destroy();
      }
      
      if (isModalOpen && editorRef.current) {
        const newEditor = new JSONEditor(editorRef.current, {
          theme: 'bootstrap4',
          disable_collapse: true,
          disable_edit_json: true,
          disable_properties: true,
          use_name_attributes: false,
          schema: {
            properties: data
          }
        });

        setEditor(newEditor);

        newEditor.on('ready', () => {
          const handleClickOutside = event => {
            const targetSelector = '#editor-modal .card .col-md-12[data-schemapath] > div [data-schematype="string"] .form-control';

            if (event.target.matches(targetSelector) || event.target.closest(targetSelector)) {
              const sourceRect = event.target.getBoundingClientRect();
              const ariaLabel = event.target.getAttribute('aria-label');
              setNode({ id: ariaLabel, message: event.target.value , require: '' });
              setTop(sourceRect.top + 30 + 'px');
              setLeft(sourceRect.left + 'px');
            }

            if (
              event.target.matches(targetSelector) ||
              event.target.closest(targetSelector) ||
              (aiRef.current && aiRef.current.contains(event.target))
            ) {
              setIsAI(true);
            } else {
              setNode({ id: null, message: '', require: '' });
              setIsAI(false);
            }
          };

          document.addEventListener('mousedown', handleClickOutside);

          return () => {
            newEditor.destroy();
            document.removeEventListener('mousedown', handleClickOutside);
          };
        });
      }
    }, 0);
  }, [data, isModalOpen]);

  
const SETTINGS = {
  itemHeight: 60,
  amount: 15,
  tolerance: 5,
  minIndex: 0,
  maxIndex: 322,
  startIndex: 0
};

const getData = (offset, limit) => {
  const start = Math.max(SETTINGS.minIndex, offset);
  const end = Math.min(offset + limit - 1, SETTINGS.maxIndex);
  const output = Object.keys(data).slice(start, end).map(key => ({ key, text: data[key] }));
  return output;
};

const rowTemplate = item => (
  <div className="item" key={item.key}>
    {
      Array.isArray(item.text) 
      ? 
        <>
          ARRAY:
          <Card>
            {item.text.map((child) => (
              <Card>
                Key: <input key={child.value} type="text" className="form-control" value={child.value} />
                Text: <input key={child.text} type="text" className="form-control" value={child.text} />
              </Card>
            ))} 
          </Card>
        </>
      : <input key={item.key} type="text" className="form-control" value={item.text} />
    }
  </div>
);

  return (
    <Modal
      title={SETTING}
      width={'90%'}
      visible={isOpen}
      onCancel={onCancel}
      style={{ maxWidth: '1200px' }}
      bodyStyle={{ overflowY: 'auto', maxHeight: 'calc(100vh - 300px)' }}
      footer={[
        <Button key="back" onClick={onCancel}>
          {CANCEL}
        </Button>,
        <Button key="danger" type="danger" onClick={onReset}>
          {RESET_TO_DEFAULT}
        </Button>,
        <Button key="submit" type="primary" onClick={onSubmit}>
          {SUBMIT}
        </Button>
      ]}
    >
      <Scroller
        get={getData}
        className="viewport"
        settings={SETTINGS}
        row={rowTemplate}
      />
      {/* <div
        ref={aiRef}
        style={{ position: 'fixed', zIndex: 1, top, left, display: isAI ? 'none' : 'none' }}
      >
        <Card title={AI_FORM} style={{ width: 300 }}>
          <Form onSubmit={onSubmitAI}>
            <Form.Item
              name="requirement"
              label={ENTER_REQUIREMENT + ' ' + node.id + ':'}
              extra={REQUIRE_PROCESSING_MESSAGE}
              rules={[{ required: true, message: REQUIRE_REQUIREMENT_MESSAGE }]}
            >
              <Input placeholder="Your requirement" onChange={onChangeAI} />
            </Form.Item>
            <Form.Item>
              <Button type="primary" htmlType="submit">
                {SUBMIT}
              </Button>
            </Form.Item>
          </Form>
        </Card>
      </div>
      <div id="editor-modal" ref={editorRef}></div> */}
    </Modal>
  );
};

export default EditorModal;
