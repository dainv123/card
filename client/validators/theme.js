import * as yup from 'yup';
import { MAX_MESSAGE, REQUIRED_MESSAGE } from '../constants/wording';

export const createThemeSchema = yup.object().shape({
    name: yup
      .string()
      .max(255, MAX_MESSAGE(255))
      .required(REQUIRED_MESSAGE('Name')),
    path: yup
      .string()
      .max(255, MAX_MESSAGE(255))
      .required(REQUIRED_MESSAGE('Path')),
  });