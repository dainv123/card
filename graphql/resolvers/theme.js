import Joi from 'joi';
import mongoose from 'mongoose';
import { Theme } from '../../models/models';
import validators from '../validators/validators';
import * as Auth from '../../helpers/auth';

export default {
  Query: {
    themes: async (root, args, context, info) => {
      const themes = await Theme.find()
        .populate('tags', 'id name')
        .exec();
      return themes;
    },
    theme: async (root, args, context, info) => {
      await Joi.validate(args, validators.theme.findTheme);
      return Theme.findById(args.id)
        .populate('tags', 'id name')
        .exec();
    },
    publicTheme: async (root, args, context, info) => {
      return Theme.findById(args.id);
    },
    publicThemes: async (root, args, context, info) => {
      const themes = await Theme.find()
        .populate('tags', 'id name')
        .exec();
      return themes;
    }
  },
  Mutation: {
    createTheme: async (root, args, context, info) => {
      await Joi.validate(args, validators.theme.createTheme, { abortEarly: false });

      const tagIds = args.tags.map(id => mongoose.Types.ObjectId(id));

      const theme = new Theme({
        name: args.name,
        path: args.path,
        tags: tagIds,
        image: args.image
      });

      const savedTheme = await theme.save();

      return savedTheme;
    },

    updateTheme: async (root, args, context, info) => {
      const themeToUpdate = await Theme.findById(args.id);

      if (!themeToUpdate) {
        throw new Error('Theme not found');
      }

      if (args.name) {
        themeToUpdate.name = args.name;
      }

      if (args.path) {
        themeToUpdate.path = args.path;
      }

      if (args.tags) {
        themeToUpdate.tags = args.tags;
      }

      if (args.image) {
        themeToUpdate.image = args.image;
      }

      const updatedTheme = await themeToUpdate.save();

      return updatedTheme;
    },

    deleteTheme: async (root, args, context, info) => {
      const themeToDelete = await Theme.findById(args.id);

      if (!themeToDelete) {
        throw new Error('Theme not found');
      }

      await themeToDelete.remove();

      return { id: args.id };
    }
  }
};
