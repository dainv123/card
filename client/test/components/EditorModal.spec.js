import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import EditorModal from '../../components/EditorModal/EditorModal';
import { SELECT_THEME, SUBMIT } from '../../constants/wording';
import { FIRST_NODE, FIRST_VALUE, THEME_PROPERTIES } from '../data/theme';
import '@testing-library/jest-dom/extend-expect'; 

describe('EditorModal component', () => {
  const mockData = THEME_PROPERTIES;
  const mockHandleOk = jest.fn();
  const mockHandleCancel = jest.fn();
  const mockHandleScrollToElement = jest.fn();

  it('should check that two input elements are nearby', () => {
    render(
      <EditorModal
        data={mockData}
        isModalOpen={true}
        handleOk={mockHandleOk}
        handleCancel={mockHandleCancel}
        handleScrollToElement={mockHandleScrollToElement}
      />
    );

    // expect(document.body).toMatchSnapshot();
    const firstInput = screen.getByDisplayValue(FIRST_NODE);
    expect(firstInput).toBeInTheDocument();
    const secondInput = firstInput.parentElement.querySelector('input:not([disabled])');
    expect(secondInput).toBeInTheDocument();
    expect(secondInput).toHaveValue(FIRST_VALUE);
  });

  it('calls handleOk when submit button is clicked', () => {
    const { getByText } = render(
      <EditorModal
        data={mockData}
        isModalOpen={true}
        handleOk={mockHandleOk}
        handleCancel={mockHandleCancel}
        handleScrollToElement={mockHandleScrollToElement}
      />
    );

    // Click the submit button
    fireEvent.click(getByText(SUBMIT));

    // Expect handleOk to be called
    expect(mockHandleOk).toHaveBeenCalled();
  });
});
