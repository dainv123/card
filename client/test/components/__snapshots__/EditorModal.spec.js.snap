// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorModal component renders with given data 1`] = `
<body
  style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask fade-appear"
      />
      <div
        aria-labelledby="rcDialogTitle0"
        class="ant-modal-wrap "
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ant-modal zoom-appear"
          role="document"
          style="max-width: 1200px; width: 90%;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <i
                  aria-label="icon: close"
                  class="anticon anticon-close ant-modal-close-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </i>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="rcDialogTitle0"
              >
                Setting
              </div>
            </div>
            <div
              class="ant-modal-body"
              style="overflow-y: auto; max-height: calc(100vh - 300px);"
            >
              <div
                class="ant-col ant-col-15"
              >
                <div
                  aria-label="grid"
                  aria-readonly="true"
                  class="ReactVirtualized__Grid ReactVirtualized__List viewport"
                  role="grid"
                  style="box-sizing: border-box; direction: ltr; height: 421px; position: relative; width: 528px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
                  tabindex="0"
                >
                  <div
                    class="ReactVirtualized__Grid__innerScrollContainer"
                    role="rowgroup"
                    style="width: auto; height: 8580px; max-width: 528px; max-height: 8580px; overflow: hidden; position: relative;"
                  >
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 0px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_11"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="img/navbar-brand.png"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 60px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_26"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="The Couple"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 120px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_32"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Events"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 180px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_38"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Love Story"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 240px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_44"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="img/navbar-brand.png"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 300px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_52"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Photos"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 360px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_58"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Wedding Gifts"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 420px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_64"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="RSVP"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 480px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_74"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="url('./images/bg-hero.jpg')"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 540px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_83"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Together with Their Families"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 600px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_89"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="images/the-couple-hero.jpg"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 660px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_95"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Milea"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 720px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_98"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="img/icon-wedding-ampersand.svg"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 780px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_101"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="Dilan"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 840px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_105"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="
                            Request the honor of your presence at their wedding ceremony and reception.
                        "
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 900px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_107"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="img/ornament-divider.png"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 960px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_126"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="The Happy Couple"
                      />
                    </div>
                    <div
                      class="item"
                      style="height: 60px; left: 0px; position: absolute; top: 1020px; width: 100%;"
                    >
                      <input
                        class="form-control input-label"
                        disabled=""
                        type="text"
                        value="node_129"
                      />
                      <input
                        class="form-control"
                        type="text"
                        value="
                            A happy marriage is the union of two good forgivers.
                        "
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-9 viewport"
              >
                No thing to select!
              </div>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn"
                type="button"
              >
                <span>
                  Cancel
                </span>
              </button>
              <button
                class="ant-btn ant-btn-danger"
                type="button"
              >
                <span>
                  Switch content background
                </span>
              </button>
              <button
                class="ant-btn ant-btn-danger"
                type="button"
              >
                <span>
                  Reset to default
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
